ls1intum/Artemis
# [Bugfix] Truncate build logs

### Checklist

### Motivation and Context


Build logs entries which are larger than 255 characters currently crash the result creation as they don't fit into the database column VARCHAR(255). This happens for example when a build fails with static code analysis enabled, as Checkstyle will output quite a large error message.

### Description

- Truncate the logs to 255 characters if they are larger than 255 characters
- Little restructure of code. Preparation of build logs moved into a private method
- Filtering of useless SCA build logs e.g. "[INFO] Downloading", "[INFO] Downloaded"
- Integration tests for various sizes of build logs

### Steps for Testing


1. Create a programming exercise (with and without SCA)
2. Participate in those exercises
3. Submit valid solutions
--> Should compile and show results
4. Submit solution containing errors
--> Should not compile and build logs should be shown

### Test Coverage

* BambooService.java: 77->80%
* BambooBuildResultNotificationDTO.BuildLogDTO 0% -> 100%
